src/semgrep_agent/meta.py

```diff
@@ -57,30 +63,46 @@ def commit(self) -> gitpython.Commit:  # type: ignore
         debug_echo(f"found commit: {commit!r}")
         return commit
 
+    @cachedproperty
+    def branch(self) -> Optional[str]:
+        try:
+            br = self.repo.active_branch.name
+        except:
+            br = None
+        return os.getenv("SEMGREP_BRANCH") or br
+
+    @cachedproperty
+    def ci_job_url(self) -> Optional[str]:
+        return os.getenv("SEMGREP_JOB_URL")
+
+    @cachedproperty
+    def pr_id(self) -> Optional[str]:
+        return os.getenv("SEMGREP_PR_ID")
+
+    @cachedproperty
+    def pr_title(self) -> Optional[str]:
+        return os.getenv("SEMGREP_PR_TITLE")
+
     def to_dict(self) -> Dict[str, Any]:
         return {
+            # REQUIRED for semgrep-app backend
             "repository": self.repo_name,
-            "ci_job_url": None,
-            "environment": self.environment,
+            #  OPTIONAL for semgrep-app backend
+            "repo_url": self.repo_url,
+            "branch": self.branch,
+            "ci_job_url": self.ci_job_url,
             "commit": self.commit_sha,
-            "commit_committer_email": self.repo.head.commit.committer.email,
-            "commit_timestamp": self.commit.committed_datetime.isoformat(),
             "commit_author_email": self.repo.head.commit.author.email,
             "commit_author_name": self.repo.head.commit.author.name,
             "commit_author_username": None,
             "commit_author_image_url": None,
-            "commit_authored_timestamp": self.commit.authored_datetime.isoformat(),
             "commit_title": self.commit.summary,
             "config": self.config,
             "on": self.event_name,
-            "branch": None,
-            "pull_request_timestamp": None,
             "pull_request_author_username": None,
             "pull_request_author_image_url": None,
-            "pull_request_id": None,
-            "pull_request_title": None,
-            "semgrep_version": sh.semgrep(version=True).strip(),
-            "python_version": sh.python(version=True).strip(),
```

Comment:
None of these are used by the semgrep_app backend, so I assumed it was ok to delete them. @underyx does that seem right to you or am I missing something?

Reply:
I thought we used the pull_request_id and pull_request_title.